python/pip_install/extract_wheels/lib/bazel.py

```diff
@@ -409,13 +409,18 @@ def extract_wheel(
     directory_path = Path(directory)
     entry_points = []
     for name, entry_point in sorted(whl.entry_points().items()):
-        entry_point_script = f"{WHEEL_ENTRY_POINT_PREFIX}_{name}.py"
-        (directory_path / entry_point_script).write_text(
+        # There is an extreme edge-case with entry_points that end with `.py`
+        # See: https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/test/java/com/google/devtools/build/lib/rules/python/PyBinaryConfiguredTargetTest.java#L174
+        entry_point_without_py = name[:-3] if name.endswith(".py") else name
```

Comment:
```suggestion
        entry_point_without_py = name.removesuffix(".py")
```

https://docs.python.org/3/library/stdtypes.html#str.removesuffix

Reply:
Incredible as it may seem `New in version 3.9.` and I would prefer this to be compatible with older versions of PY3.

Reply:
Bah, so it is.